tests/equivalence.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+from qiskit import QuantumCircuit, transpile
+from qiskit.quantum_info import Statevector, Operator
+
+def remove_measurements(circuit: QuantumCircuit):
+        """
+        helper: remove measurements from quantum circuit
+
+        args:
+            circuit: QuantumCircuit
+                the input quantum circuit
+        returns:
+            QuantumCircuit: the circuit without the measurements
+        """
+        new_circ = QuantumCircuit(*circuit.qregs)
+        for instruction in circuit.data:
+            if instruction.operation.name != "measure":
+                new_circ.append(instruction.operation, instruction.qubits, instruction.clbits)
+        return new_circ
+
+
+def circuits_equal_exact(qasm1: QuantumCircuit, qasm2: QuantumCircuit):
+    """
+    compare two quantum circuits to check if they are equal to each other,
+    up to some global phase
+
+    this piggybacks on the qiskit Operation's equiv(), which finds the entire unitary
+    of the circuit. thus, it's really slow and memory-intensive, as the matrices
+    will be 2^n by 2^n, but it is exact
+
+    try to use with <= 12 qubits. anything over that might throw due to memory
+    issues
+
+    args:
+        qasm1, qasm2: QuantumCircuit
+            the input quantum circuits to check
+    returns:
+        bool: True if equal, False ow
+    """
+    if qasm1.num_qubits != qasm2.num_qubits:
+        return False
+    c1 = transpile(remove_measurements(qasm1), basis_gates=['u3', 'cx'])
+    c2 = transpile(remove_measurements(qasm2), basis_gates=['u3', 'cx'])
```

Comment:
Also, we don't want to transpile the circuit so this line and the one above can be removed.